core/io/file_access_memory.h

```diff
@@ -32,38 +32,46 @@
 #define FILE_ACCESS_MEMORY_H
 
 #include "core/io/file_access.h"
+#include "core/templates/safe_refcount.h"
 
 class FileAccessMemory : public FileAccess {
+	struct FileInfo {
+		PackedByteArray data;
```

Comment:
Could we expand this `struct` to also represent `uint8_t *data` so we can combine and simplify the implementation?